### ISSUE TITLE
s3s-fs test doesn't compile & issue with connector

### ISSUE THREAD

**Issue by TheMenko** [Nugine/s3s#89]
Hi, i can't get tests to compile since atoi is failing for some reason. 
Also lazy_static & sharded_slab doesn't compile but i figured that has to do something with rust 1.74 version, but i don't have time to investigate.   
I did copy the code to my local project for testing but then the connector is failing with this:
`the trait tower_service::Service<http::request::Request<SdkBody>> is not implemented for Connector`.
It seems that it doesn't figure out from where SdkBody is comming from because it acknowledges that there is an impl for  
`aws_smithy_http::body::SdkBody`

Basically this part is failing only:
```
     // Convert to aws http connector
     let conn = Connector::from(service.into_shared());

     // Setup aws sdk config
     SdkConfig::builder()
         .credentials_provider(SharedCredentialsProvider::new(cred))
         .http_connector(conn) // <--- this right here
         .region(Region::new(REGION))
         .endpoint_url(format!("http://{DOMAIN}"))
         .build()

```

**Comment by TheMenko** [comment]
Update: this seems to be some change in aws sdk, since i used main branch instead of version.
Setting fixed version to 0.31.1 fixes the problem

**Comment by Nugine** [comment]
The "trait not implemented" problem is mostly caused by incompatible dependency versions.
It seems that your aws sdk version is incompatible with `s3s-aws`'s dependencies since aws sdk bumps version frequently.
I'm not sure whether there is anything I can do here.

**Comment by TheMenko** [comment]
Closing this, thanks
